impl/src/main/java/org/ehcache/internal/store/heap/SerializedOnHeapValueHolder.java

```diff
@@ -15,24 +15,23 @@
  */
 package org.ehcache.internal.store.heap;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import org.ehcache.exceptions.SerializerException;
 import org.ehcache.spi.cache.Store;
 import org.ehcache.spi.serialization.Serializer;
 
-import java.io.IOException;
+import java.lang.ref.SoftReference;
 import java.nio.ByteBuffer;
 
-class ByValueOnHeapValueHolder<V> extends OnHeapValueHolder<V> {
+class SerializedOnHeapValueHolder<V> extends OnHeapValueHolder<V> {
   private final ByteBuffer buffer;
-  private final int hash;
   private final Serializer<V> serializer;
 
-  protected ByValueOnHeapValueHolder(V value, long creationTime, Serializer<V> serializer) {
-    this(value, creationTime, NO_EXPIRE, serializer);
-  }
+  @SuppressFBWarnings("SE_BAD_FIELD")
+  private SoftReference<V> cachedValue;
```

Comment:
I understand the goal here but want to draw attention to this construct.
Given the places where `getCurrentValue` is called, it will most likely be _always_ populated, which may have a negative impact on the memory footprint of the `OnHeapStore` with a `SerializingCopier`.
Opinions?